x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_migration_client.ts

```diff
@@ -45,8 +53,80 @@ export class RuleMigrationsDataMigrationClient extends RuleMigrationsDataBaseCli
         return this.processHit(document);
       })
       .catch((error) => {
+        if (isStringValidJSON(error.message)) {
+          const message = JSON.parse(error.message);
+          if (Object.hasOwn(message, 'found') && message.found === false) {
```

Comment:
Can we encapsulate `isNotFoundError` instead of `isStringValidJSON`? So we parse the error only once, and we keep this implementation cleaner.

Reply:
This has been done as well. Please re-review.